### DISCUSSION TITLE
Hardware recognition

### DISCUSSION THREAD

**Discussion by Bumblebee001** [GrandOrgue/grandorgue#1635] [General]
I noticed that a keyboard attached to a USB hub is **_not detected_** by GO (Listen to event). Basically, the keyboard has to be connected directly to the computer to be detectable. 

A similar problem has also been experienced by a friend whose hardware setup involves the use of a multiport USB hub to be able to connect everything to the PC. In his case his GO was able to detect the keyboards etc but he had to manually assign a numerical identity to get them working correctly.

In my case, the laptop has only one USB port so almost inevitably a USB hub becomes necessary. The good thing is that I use the laptop for testing purposes so I only need to unplug the hub and plug the keyboard instead. But if I would be using this same laptop with several keyboards, then there is likely to be a problem.

I am not sure if this is similar to #1626 / #1616.

**Comment by oleg68** [comment]
1. What OS are you using?
2. Are the devices present in File -> Settings -> Midi Devices?

**Comment by Bumblebee001** [comment]
On Wed, 16 Aug 2023 at 08:20, Oleg Samarin ***@***.***> wrote:

>
>    1. What OS are you using?
>    2. Are the devices present in File -> Settings -> Midi Devices?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/GrandOrgue/grandorgue/discussions/1635#discussioncomment-6737162>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVM7NW47XF6YVBVGJTGBJ3XVRRENANCNFSM6AAAAAA3RWOIXA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
Windows 11

Whether or not the device is present, normally when I plug in the keyboard,
I right click on the virtual keyboard and click ‘listen to event’. Nothing
happens if the keyboard is plugged into the hub but it is recognizes if
plugged directly into the laptop.

**Reply by oleg68** [reply]
@Bumblebee001 

> Whether or not the device is present, normally when I plug in the keyboard,
I right click on the virtual keyboard and click ‘listen to event’. 

This answer does not help me to troubleshoot your problem. The key is on the `MIDI Devices` settings. Please post there two screenshots of you `MIDI Devices` when everything works and when some devices do not work.

**Comment by lewisalexander2020** [comment]
some useful questions to ask here…

1: make / model of USB hub?
2: Is USB Hub powered or USB power only?
3: is the USB socket of your laptop USB3 / USB-C?
4: the “keyboard/s you mention, make and model please if possible, or are you relying on a controller interface from someone like midiboutique connected to a fatar keybed?

These questions help to trace a problem of whether a device is receiving enough power to not only be recognised, but also to work at it’s full capacity, 

my own setup, I use a 2020 16” Macbook Pro, I use 2 x 4 port USB3 bus powered hubs with USB-C adaptors running 4 manuals on hub 1, pedalboard and 3 switch scanner boards on hub 2 and they run fine in the developer job I’m working on. Haven’t checked with GO as accessibility with a screen reader is mostly difficult as heck, but just trying to demonstrate that if you can isolate certain conditions and give some further detail, we can see where you’re coming from.

good example here, this is my device chain:

USB-HUB_1:
-MKCV64SMFU (1) midi channel 1 for manual 1
-MKCV64SMFU (2) midi channel 2 for manual 2
-MKCV64SMFU (3) midi channel 3 for manual 3
-MKCV64SMFU (4) midi channel 4 for manual 4

USB-HUB_2:
-MKC64U (1) Midi Channel 5 for Pedalboard
-MKC64U (2) Midi Channel 6 Left stop jamb units
-MKC64U (3) Midi Channel 7 Right stop jamb units
-MKC64U (4) Midi Channel 8 pistons / control functions to console

lew

**Comment by Bumblebee001** [comment]
1. Vention
2. Usb power
3. USB 3.2 type A port
4. M-key v2 with cable that plugs directly into the USB port

**Comment by lewisalexander2020** [comment]
is this keyboard responding with other software without any issues, such as a DAW or other software instrument? if so, it might be how GO sees external devices such as hub connected midi devices.

lew

**Comment by Bumblebee001** [comment]
I tried jOrgan… ok tried grandorgue again and it is being detected now. How
about that?

**Comment by lewisalexander2020** [comment]
It might be because the Windows MIDI and Audio System registered the device to recognise it and it’s kicked it in to the device MIDI list for GO.

I’m not a windows guy, but as a mac guy, I’ve seen this happen.

lew
